### PR TITLE
add any_perm parameter for PermissionRequiredMixin

### PR DESCRIPTION
**Pull request by 1vank1n**

### CODE REVIEW

**Comment by ad-m** [comment]
Why not adjust ```get_40x_or_None```? 

Missing tests.

**Comment by 1vank1n** [comment]
> Why not adjust get_40x_or_None?
Just have tried to "Keep It Stupid Simple".

> Missing tests.
Yeap. If it could be potential merged, I'll write them.

**Comment by ad-m** [comment]
I understand that it was easy to make such a change without analyzing the whole thing, but it is more difficult to trace the code and maintain it in such a way, because similar logic is spread in two places.

**Comment by 1vank1n** [comment]
Ok! I'll rewrite to adjust `get_40x_or_None`

**Comment by 1vank1n** [comment]
@ad-m what about this build?

File: guardian/utils.py [hunk 1]
```diff
@@ -106,8 +107,12 @@ def get_40x_or_None(request, perms, obj=None, login_url=None,
...
     # if still no permission granted, try obj perms
     if not has_permissions:
-        has_permissions = all(request.user.has_perm(perm, obj)
-                              for perm in perms)
+        if any_perm:
+            has_permissions = any(request.user.has_perm(perm, obj)
+                                for perm in perms)
+        else:
+            has_permissions = all(request.user.has_perm(perm, obj)
+                                for perm in perms)
```
Comment (michael-k) [inline guardian/utils.py:115]:
```suggestion
            has_permissions = any(request.user.has_perm(perm, obj)
                                  for perm in perms)
        else:
            has_permissions = all(request.user.has_perm(perm, obj)
                                  for perm in perms)
```

**Pull request merged**